Tools/ardupilotwaf/configuration_helper.py

```diff
@@ -0,0 +1,78 @@
+"""
+WAF Tool that helps on configuration parameter, creating the AP_Config
+header file.
+
+This tool needs compiler_c and compiler_cxx to be loaded, make sure
+you load them before this tool.
+
+Example::
+    def configure(cfg):
+        cfg.load('configuration_helper')
+"""
+
+def configure(cfg):
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isfinite(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISFINITE",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isinf(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISINF",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <cmath>
+
+                  int main() {
+                    return std::isnan(1.0f);
+                  }''',
+                  define_name="HAVE_CMATH_ISNAN",
+                  mandatory=False)
+
+    cfg.check_cxx(fragment='''
+                  #include <math.h>
+                  #include <cmath>
+
+                  using std::isfinite;
+
+                  int main() {
+                    return isfinite((double)1);
+                  }''',
+                  define_name="NEED_CMATH_ISFINITE_STD_NAMESPACE",
+                  cxxflags="-std=gnu++11",
```

Comment:
@OXINARF Hm, yeah, I think what you're saying makes sense. I mean, it makes more sense to use flags we define statically (i.e., those set by `board.configure_env`) to run the checks. I think we should be fine with placing `cfg.load('configuration_helper')` at the end of `Board.configure()` and removing from `toolchain.py`.

@ricardotk I just tested that approach with `arducopter-quad` and that seems to work fine here. With that we don't need to explicitly set `-std=gnu++11` and we'll have all the other "static" flags being used.